coderd/users_test.go

```diff
@@ -1721,6 +1722,83 @@ func TestSuspendedPagination(t *testing.T) {
 	require.Equal(t, expected, page.Users, "expected page")
 }
 
+func TestUserAutofillParameters(t *testing.T) {
+	t.Parallel()
+	t.Run("NotSelf", func(t *testing.T) {
+		t.Parallel()
+		client, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+
+		u1 := coderdtest.CreateFirstUser(t, client)
+
+		_, u2 := coderdtest.CreateAnotherUser(t, client, u1.OrganizationID)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		db := api.Database
+
+		version := dbfake.TemplateVersion(t, db).Seed(database.TemplateVersion{
+			CreatedBy:      u1.UserID,
+			OrganizationID: u1.OrganizationID,
+		}).Params(database.TemplateVersionParameter{
+			Name:     "param",
+			Required: true,
+		}).Do()
+
+		_, err := client.UserAutofillParameters(
+			ctx,
+			u2.ID.String(),
+			version.Template.ID.String(),
+		)
+
+		var apiErr *codersdk.Error
+		require.ErrorAs(t, err, &apiErr)
+		require.Equal(t, http.StatusForbidden, apiErr.StatusCode())
+	})
+
+	t.Run("FindsParameters", func(t *testing.T) {
+		t.Parallel()
+		client, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+
+		u1 := coderdtest.CreateFirstUser(t, client)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		db := api.Database
+
+		version := dbfake.TemplateVersion(t, db).Seed(database.TemplateVersion{
+			CreatedBy:      u1.UserID,
+			OrganizationID: u1.OrganizationID,
+		}).Params(database.TemplateVersionParameter{
+			Name:     "param",
+			Required: true,
+		}).Do()
+
+		coderdtest.CreateWorkspace(t, client, u1.OrganizationID, version.Template.ID,
+			func(cwr *codersdk.CreateWorkspaceRequest) {
+				cwr.RichParameterValues = []codersdk.WorkspaceBuildParameter{
+					{
+						Name:  "param",
+						Value: "foo",
+					},
+				}
+			})
+
+		params, err := client.UserAutofillParameters(
+			ctx,
+			u1.UserID.String(),
+			version.Template.ID.String(),
+		)
+		require.NoError(t, err)
+
+		require.Equal(t, 1, len(params))
+
+		require.Equal(t, "param", params[0].Name)
+		require.Equal(t, "foo", params[0].Value)
```

Comment:
I would like to see a test case with multiple builds and verify that we return the latest parameters.

Reply:
I'd also propose adding builds from a separate user and validating that none of that user's parameter values show up when we query parameters for a separate user.